src/libopenrave/kinbodylink.cpp

```diff
@@ -212,24 +212,27 @@ void KinBody::LinkInfo::SerializeJSON(rapidjson::Value &value, rapidjson::Docume
         value.AddMember("readableInterfaces", std::move(rReadableInterfaces), allocator);
     }
 
-    // if(_mapExtraGeometries.size() > 0 ) {
-    //     rapidjson::Value extraGeometriesValue;
-    //     extraGeometriesValue.SetObject();
-    //     FOREACHC(im, _mapExtraGeometries) {
-    //         rapidjson::Value geometriesValue;
-    //         geometriesValue.SetArray();
-    //         FOREACHC(iv, im->second){
-    //             if(!!(*iv))
-    //             {
-    //                 rapidjson::Value geometryValue;
-    //                 (*iv)->SerializeJSON(geometryValue, allocator);
-    //                 geometriesValue.PushBack(geometryValue, allocator);
-    //             }
-    //         }
-    //         extraGeometriesValue.AddMember(rapidjson::Value(im->first.c_str(), allocator).Move(), geometriesValue, allocator);
-    //     }
-    //     value.AddMember("extraGeometries", extraGeometriesValue, allocator);
-    // }
+    if (_mapExtraGeometries.size() > 0) {
+        rapidjson::Value extraGeometriesValue;
+        extraGeometriesValue.SetArray();
+        FOREACHC(im, _mapExtraGeometries) {
+            rapidjson::Value extraGeometryValue;
+            extraGeometryValue.SetObject();
+            orjson::SetJsonValueByKey(extraGeometryValue, "name", im->first, allocator);
```

Comment:
I think you missed handling the `id` field here and in deserialization

Reply:
thanks for offline conversation, I think you can put `id` instead of `name` here.
the `key` of the `std::map` is now represented by `id`.